### PR TITLE
Update to resolve CI warnings

### LINKED ISSUE #2649: Fix CI warnings
```

tests/ignite/handlers/test_state_param_scheduler.py: 15 warnings
  /opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/torch/testing/_deprecated.py:35: FutureWarning: torch.testing.assert_allclose() is deprecated since 1.12 and will be removed in 1.14. Use torch.testing.assert_close() instead. For detailed upgrade instructions see https://github.com/pytorch/pytorch/issues/61844.
    warnings.warn(msg, FutureWarning)

tests/ignite/metrics/test_classification_report.py: 48 warnings
  /opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in samples with no predicted labels. Use `zero_division` parameter to control this behavior.
    _warn_prf(average, modifier, msg_start, len(result))

tests/ignite/metrics/test_classification_report.py: 48 warnings
  /opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Recall and F-score are ill-defined and being set to 0.0 in samples with no true labels. Use `zero_division` parameter to control this behavior.
    _warn_prf(average, modifier, msg_start, len(result))

tests/ignite/metrics/test_precision.py: 40 warnings
tests/ignite/metrics/test_recall.py: 40 warnings
  /home/runner/work/ignite/ignite/ignite/metrics/precision.py:55: RuntimeWarning: `y` and `y_pred` should be of dtype long when entry type is binary and average!=False
    "`y` and `y_pred` should be of dtype long when entry type is binary and average!=False", RuntimeWarning

tests/ignite/metrics/test_precision.py: 30 warnings
  /opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
    _warn_prf(average, modifier, msg_start, len(result))

tests/ignite/metrics/test_precision.py: 10 warnings
  /opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 due to no predicted samples. Use `zero_division` parameter to control this behavior.
    _warn_prf(average, modifier, msg_start, len(result))

tests/ignite/metrics/gan/test_inception_score.py::test_inception_score
tests/ignite/metrics/gan/test_utils.py::test_inception_extractor_wrong_inputs
tests/ignite/metrics/gan/test_utils.py::test_inception_model_probability
  /opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/torchvision/models/_utils.py:209: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and will be removed in 0.15, please use 'weights' instead.
    f"The parameter '{pretrained_param}' is deprecated since 0.13 and will be removed in 0.15, "

tests/ignite/metrics/gan/test_inception_score.py::test_inception_score
tests/ignite/metrics/gan/test_utils.py::test_inception_extractor_wrong_inputs
tests/ignite/metrics/gan/test_utils.py::test_inception_model_probability
  /opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and will be removed in 0.15. The current behavior is equivalent to passing `weights=Inception_V3_Weights.IMAGENET1K_V1`. You can also use `weights=Inception_V3_Weights.DEFAULT` to get the most up-to-date weights.
    warnings.warn(msg)
```

### PR DESCRIPTION
**Pull request by puhuk**
Fixes #2649

Description: 
Update to resolve CI warnings
Will send soon to fix warnings in `test_precision.py`

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/gan/utils.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def __init__(self, return_features: bool, device: Union[str, torch.device] = "cp
             raise RuntimeError("This module requires torchvision to be installed.")
         super(InceptionModel, self).__init__()
         self._device = device
-        self.model = models.inception_v3(pretrained=True).to(self._device)
+        self.model = models.inception_v3(weights=models.Inceaaaption_V3_Weights.DEFAULT).to(self._device)
```
Comment (vfdev-5) [inline ignite/metrics/gan/utils.py:26]: Here is a typo in `Inceaaaption_V3_Weights`. Another point here is that this will fail with old pytorch version and min supported version is 1.3.1. See https://github.com/pytorch/ignite/actions/runs/2864338520

File: tests/ignite/handlers/test_state_param_scheduler.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ def test_pwlinear_scheduler_step_constant(max_epochs, milestones_values):
     )
     linear_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
     engine.run([0] * 8, max_epochs=max_epochs)
-    torch.testing.assert_allclose(getattr(engine.state, "pwlinear_scheduled_param"), milestones_values[0][1])
+    torch.testing.assert_close(getattr(engine.state, "pwlinear_scheduled_param"), float(milestones_values[0][1]))
```
Comment (vfdev-5) [inline tests/ignite/handlers/test_state_param_scheduler.py:98]: Same here about pytorch min supported version and `torch.testing.assert_close`

File: ignite/metrics/gan/utils.py (hunk 1 above)
Comment (puhuk) [reply ignite/metrics/gan/utils.py:26]: Sorry for the typo. Let me correct this.
For minimum pytorch version,  may I ask what is typical methods to deal with this.
Is this ok with condition like this?

```
if torch.__version__ < 1.3.1:
    ~~~~
```

Comment (vfdev-5) [reply ignite/metrics/gan/utils.py:26]: Here is the way we check for torch version:
https://github.com/pytorch/ignite/blob/3d9ba50bba42438a63183b3db0d4cc3a5e9b01ac/ignite/metrics/gan/fid.py#L197

And to clarify, our code needs to support pytorch >= 1.3.1, for below we do not care anymore.

File: ignite/metrics/gan/utils.py [hunk 3]
```diff
@@ -23,7 +24,10 @@ def __init__(self, return_features: bool, device: Union[str, torch.device] = "cp
             raise RuntimeError("This module requires torchvision to be installed.")
         super(InceptionModel, self).__init__()
         self._device = device
-        self.model = models.inception_v3(pretrained=True).to(self._device)
+        if Version(torch.__version__) <= Version("1.7.0"):
+            self.model = models.inception_v3(pretrained=True).to(self._device)
+        else:
+            self.model = models.inception_v3(weights=models.Inception_V3_Weights.DEFAULT).to(self._device)
```
Comment (vfdev-5) [inline ignite/metrics/gan/utils.py:30]:
```suggestion
        if Version(torch.__version__) <= Version("1.7.0"):
            model_kwargs = {"pretrained": True}
        else:
            model_kwargs = {"weights": models.Inception_V3_Weights.DEFAULT}
        self.model = models.inception_v3(model_kwargs).to(self._device)
```

File: tests/ignite/handlers/test_state_param_scheduler.py [hunk 4]
```diff
@@ -95,7 +96,10 @@ def test_pwlinear_scheduler_step_constant(max_epochs, milestones_values):
     )
     linear_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
     engine.run([0] * 8, max_epochs=max_epochs)
-    torch.testing.assert_allclose(getattr(engine.state, "pwlinear_scheduled_param"), milestones_values[0][1])
+    if Version(torch.__version__) <= Version("1.7.0"):
+        torch.testing.assert_allclose(getattr(engine.state, "pwlinear_scheduled_param"), milestones_values[0][1])
+    else:
+        torch.testing.assert_close(getattr(engine.state, "pwlinear_scheduled_param"), float(milestones_values[0][1]))
```
Comment (vfdev-5) [inline tests/ignite/handlers/test_state_param_scheduler.py:102]: Instead of repeating this if/else code N times, you can once define in the global context a function `torch_testing_assert_close` : 
```python
if Version(torch.__version__) <= Version("1.7.0"):
    torch_testing_assert_close = torch.testing.assert_allclose
else:
    torch_testing_assert_close = torch.testing.assert_close
```
and use it in each tests.

See for example here a similar idea :  https://github.com/pytorch/ignite/commit/d8af03c2e34e8dff22e17570a32aedd0b755ff82

File: tests/ignite/handlers/test_state_param_scheduler.py [hunk 5]
```diff
@@ -95,7 +96,11 @@ def test_pwlinear_scheduler_step_constant(max_epochs, milestones_values):
     )
     linear_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
     engine.run([0] * 8, max_epochs=max_epochs)
-    torch.testing.assert_allclose(getattr(engine.state, "pwlinear_scheduled_param"), milestones_values[0][1])
+    if Version(torch.__version__) <= Version("1.7.0"):
+        torch_testing_assert_close = torch.testing.assert_allclose
+    else:
+        torch_testing_assert_close = torch.testing.assert_close
+    torch_testing_assert_close(getattr(engine.state, "pwlinear_scheduled_param"), float(milestones_values[0][1]))
```
Comment (vfdev-5) [inline tests/ignite/handlers/test_state_param_scheduler.py:103]: As I suggested previously, the point is to define `torch_testing_assert_close` **once** in the global scope

**Comment by puhuk** [comment]
For warnings from `tests/ignite/metrics/test_precision.py` as I think those warnings are intended to warn.
In my thought, to fix those warnings, corner cases like `(torch.zeros(size=(10,)), torch.randint(0, 2, size=(10,)), 1)` should be deleted.
Is that correct? Or is there other methods to fix those warnings?

**Comment by vfdev-5** [comment]
> For warnings from tests/ignite/metrics/test_precision.py as I think those warnings are intended to warn.
In my thought, to fix those warnings, corner cases like (torch.zeros(size=(10,)), torch.randint(0, 2, size=(10,)), 1) should be deleted.

Can you point to the code what you are talking about ?

**Comment by puhuk** [comment]
> > For warnings from tests/ignite/metrics/test_precision.py as I think those warnings are intended to warn.
> > In my thought, to fix those warnings, corner cases like (torch.zeros(size=(10,)), torch.randint(0, 2, size=(10,)), 1) should be deleted.
> 
> Can you point to the code what you are talking about ?

For this line, because it generates data with zeros, 
https://github.com/pytorch/ignite/blob/c506c311c95e8806e2518bf47ce3422cddedfab6/tests/ignite/metrics/test_precision.py#L154

it occurs error with being set to zero in https://github.com/pytorch/ignite/blob/c506c311c95e8806e2518bf47ce3422cddedfab6/tests/ignite/metrics/test_precision.py#L130.

When I add `zero_divion=1` like this `precision_score(np_y, np_y_pred, average=sk_average_parameter, zero_division=1)`, it occurs error when comparing computed metric and reference.

**Comment by vfdev-5** [comment]
@puhuk here is the link on why this corner case was added : https://github.com/pytorch/ignite/issues/2007
So, we need to keep this test case. Why you set `zero_division=1`, have you tried `zero_division=0` ?

**Comment by puhuk** [comment]
> 

No, I didn't because document says the default value is `warn` and it acts as `0` in https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html
But when I set with `zero_division=0`, those warning fixed.

File: tests/ignite/metrics/test_precision.py [hunk 6]
```diff
@@ -151,8 +153,8 @@ def get_test_cases():
             (torch.randint(0, 2, size=(50, 12, 10)), torch.randint(0, 2, size=(50, 12, 10)), 16),
             (torch.randint(0, 2, size=(50, 1, 12, 10)), torch.randint(0, 2, size=(50, 1, 12, 10)), 16),
             # Corner case with all zeros predictions
-            (torch.zeros(size=(10,)), torch.randint(0, 2, size=(10,)), 1),
-            (torch.zeros(size=(10, 1)), torch.randint(0, 2, size=(10, 1)), 1),
+            (torch.zeros(size=(10,)).long(), torch.randint(0, 2, size=(10,)), 1),
+            (torch.zeros(size=(10, 1)).long(), torch.randint(0, 2, size=(10, 1)), 1),
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_precision.py:157]:
```suggestion
            (torch.zeros(size=(10,), dtype=torch.long), torch.randint(0, 2, size=(10,)), 1),
            (torch.zeros(size=(10, 1), dtype=torch.long), torch.randint(0, 2, size=(10, 1)), 1),
```

**Comment by vfdev-5** [comment]
@puhuk please fix code formatting with https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md#formatting-code
```
bash ./tests/run_code_style.sh fmt
```

**Review by vfdev-5 — APPROVED** [review]
lgtm, thanks

**Comment by vfdev-5** [comment]
@puhuk actually, there is still warning about model weights : https://github.com/pytorch/ignite/runs/7903177538?check_suite_focus=true#step:15:1850

**Comment by puhuk** [comment]
Still some warings in https://github.com/pytorch/ignite/runs/7963618423?check_suite_focus=true#step:15:1826
Is it can be resolved in ignite?

**Comment by vfdev-5** [comment]
> Still some warings in https://github.com/pytorch/ignite/runs/7963618423?check_suite_focus=true#step:15:1826 Is it can be resolved in ignite?

No, these warnings are in installed packages, we can't do anything with them.

File: ignite/metrics/gan/utils.py [hunk 7]
```diff
@@ -23,7 +24,12 @@ def __init__(self, return_features: bool, device: Union[str, torch.device] = "cp
             raise RuntimeError("This module requires torchvision to be installed.")
         super(InceptionModel, self).__init__()
         self._device = device
-        self.model = models.inception_v3(pretrained=True).to(self._device)
+        if Version(torch.__version__) <= Version("1.7.0"):
+            model_kwargs = {"pretrained": True}
+            self.model = models.inception_v3(model_kwargs).to(self._device)
+        else:
+            self.model = models.inception_v3(weights=models.Inception_V3_Weights.DEFAULT).to(self._device)
```
Comment (vfdev-5) [inline ignite/metrics/gan/utils.py:31]: @puhuk this looks bad and the code wont work for True-branch of `if Version(torch.__version__) <= Version("1.7.0")`.
I previously suggested to have a single call:
```python
self.model = models.inception_v3(**model_kwargs).to(self._device)
```
Can you make this ?

**Review by vfdev-5 — APPROVED** [review]
Thanks @puhuk

File: ignite/metrics/gan/utils.py [hunk 8]
```diff
@@ -23,7 +24,13 @@ def __init__(self, return_features: bool, device: Union[str, torch.device] = "cp
             raise RuntimeError("This module requires torchvision to be installed.")
         super(InceptionModel, self).__init__()
         self._device = device
-        self.model = models.inception_v3(pretrained=True).to(self._device)
+        if Version(torch.__version__) <= Version("1.7.0"):
```
Comment (vfdev-5) [inline ignite/metrics/gan/utils.py:27]: This version check is incorrect

Comment (vfdev-5) [reply ignite/metrics/gan/utils.py:27]: You have to check for torchvision version here
